src/wallet/walletdb.cpp

```diff
@@ -770,6 +770,19 @@ static DBErrors LoadMinVersion(CWallet* pwallet, DatabaseBatch& batch) EXCLUSIVE
     return DBErrors::LOAD_OK;
 }
 
+static DBErrors LoadWalletFlags(CWallet* pwallet, DatabaseBatch& batch) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    AssertLockHeld(pwallet->cs_wallet);
+    uint64_t flags;
+    if (batch.Read(DBKeys::FLAGS, flags)) {
+        if (!pwallet->LoadWalletFlags(flags)) {
+            pwallet->WalletLogPrintf("Error reading wallet database: Unknown non-tolerable wallet flags found\n");
+            return DBErrors::TOO_NEW;
```

Comment:
In 040fa33f:

Bad copy here, error is `DBErrors::CORRUPT`.

Reply:
The original error is incorrect, it's supposed to be `TOO_NEW`.